amo/blockchain/hibernate.go

```diff
@@ -127,7 +143,10 @@ func (m MissRuns) UpdateMissRuns(height int64, vals []crypto.Address) (doValUpda
 		batch.Set(r, buf)
 	}
 
-	// TODO: remove too old runs
+	// delete finishedRuns
+	for _, r := range finishedRuns {
+		batch.Delete(r)
+	}
```

Comment:
No need to wait until this point. If the `rewind guard` is something like `range guard` then we should delete old runs there.